feat: the function of adding and deleting nodes in the outline tree, support undo.

1degrees opened the pull request:
the function of adding and deleting nodes in the outline tree, support undo.
（大纲树增加删除功能，支持撤销）

English | [简体中文](https://github.com/opentiny/tiny-engine/blob/develop/.github/PULL_REQUEST_TEMPLATE/PULL_REQUEST_TEMPLATE.zh-CN.md)

# PR

## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our [Commit Message Guidelines](https://github.com/opentiny/tiny-engine/blob/develop/CONTRIBUTING.md)
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)
- [x] Built its own designer, fully self-validated

## PR Type

What kind of change does this PR introduce?

<!-- Please check the one that applies to this PR using "x". -->

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## Background and solution
<!--
1. Describe the problem and the scenario.
2. New features need to be described and attached with renderings.
3. Screenshots or GIFs involving UI/Interaction changes/Bugfix before and after modification are required.
-->

### What is the current behavior?

<!-- Please describe the current behavior that you are modifying, or link to a relevant issue. -->

Issue Number: N/A

### What is the new behavior?


## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

<!-- If this PR contains a breaking change, please describe the impact and migration path for existing applications below. -->

## Other information


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **Chores**
  - Removed obsolete linting configurations to streamline development processes.

- **New Features**
  - Upgraded the demo package version for improved performance and new capabilities.

- **Documentation**
  - Added extensive new documentation covering user guides, developer tutorials, and practical use cases.
  - Introduced detailed walkthroughs on application creation, designer customization, block management, AI integrations, and API usage.
  - Expanded content on workflows, internationalization, and deployment options to help users navigate and leverage the platform efficiently.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

hexqi committed:
chore: update version v2.3.0-rc.0 (#1130)

Commit:
feat: 大纲树添加删除节点功能

gene9831 committed:
fix: title content lost when dragging from outline tree (#1149)

修复拖拽Header组件会导致内容消失的 bug

SonyLeo committed:
fix: insert logic and multi node insert issue (#1150)

xuanlid committed:
fix: style optimization (#1162)

修改样式问题
1.区块设置->事件设置 去掉无必要滚动条显示
2.数据源下拉校验错位

gene9831 committed:
fix: the list in ViewerSwitcher does not close after clicking the canvas (#1169)

修复嵌套路由预览下拉列表弹出后，无法通过点击画布内部来关闭的 bug

gene9831 committed:
feat: clicking node in the outline tree is invalid when page unlocked (#1172)

允许页面未锁定时通过点击大纲树 hover 和 选中节点。

chilingling committed:
fix: stop multiple js error modal popup (#1175)

lichunn committed:
fix: fix incompatible styles in dark theme (#1174)

* fix: 修复暗色主题部分不兼容的样式

xuanlid committed:
fix: java backend click on block with err message (#1177)

处理区块字段 assets为undefined的情况

xuanlid committed:
fix: Modify the dark theme style (#1186)

暗色主题下样式调整
1.下拉框面板背景
2.折叠面板图标
3.消息提示文字颜色
4.固定面板图标
5.按钮hover背景

lichunn committed:
fix: the default page settings under the folder do not take effect (#1182)

lichunn committed:
fix: add test cases for routing code output (#1189)

lichunn committed:
fix: page and block preview failure (#1187)

* fix: 修复页面区块预览失效问题

* fix:fix review

SonyLeo committed:
fix: undo or reply to save button tag display logic optimization (#1185)

SonyLeo committed:
fix: refactor multi-node click and fix outline tree click problem (#1166)

hexqi committed:
chore: update version v2.3.0 (#1192)

chilingling commented:
@gene9831 Please help with the review.

gene9831 committed:
docs: add documentation (#917)

1degrees committed:
修复数据源绑定时，将数据源初始化的State 和原有State错误更新，导致功能无效，Schame沉余无用数据 (#1208)

Co-authored-by: 张啸 <xiaozhang20@iflytek.com>
修复: 数据源绑定时，将数据源初始化的State 和原有State错误更新，导致功能无效，Schame沉余无用数据

webzmj530 committed:
fix:The packages field is lost after splitMaterials (#1200)

* fix:拆包之后packages字段丢失

* feat:合并物料时将package信息写入bundle.json

---------

Co-authored-by: mjzhu4 <mjzhu4@kxdigit.com>

gene9831 commented:
可以在pr comment中补充个图片或者gif，展示效果

chilingling committed:
fix: datasourceMap is empty when init set schema (#1142)

chilingling committed:
fix: canvas actionbar move child can not work (#1180)

chilingling committed:
fix: compact schema missing css field (#1179)

chilingling commented:
@1degrees  git commit 没有正确配置成 GitHub 用户，看看要不要修改一下 。（配置正确的话，后续可以通过代码的 git comit 记录索引到对应的GitHub用户）
<img width="791" alt="image" src="https://github.com/user-attachments/assets/3f1fc766-9ab7-4607-b680-0bb8dd7deb6a" />

Ljhhhhhh committed:
fix: 修复文案为空时 i18n 下的文案为 null 导致出码异常 (#1211)

修复文案为空时 i18n 下的文案为 null 导致出码异常，新增判断条件，默认为空对象

webzmj530 committed:
fix:When splitting materials, a component can exist in multiple snippets. Also, there can be two components within one snippet (with the same componentName but different snippetName) (#1212)

* 拆包的时候一个组件在多个分组里面也可以一个分组里面存在两个组件(componentName相同, snippetName不同)

SonyLeo committed:
fix: the display is abnormal when clicking back on multiple selected nodes (#1201)

1. 多选节点进行点击回退剩余一个元素时，框选效果异常
2. 快捷键剪切失效

xuanlid committed:
feat: Modify the outer class name of the module (#1181)

1. 修改plugin下模块外层类名
2. 复选框选中状态和默认状态大小调整
3. 部分模块滚动条修改
4. 弹窗关闭图标样式调整

xuanlid committed:
feat: Modify the theme switching logic and  add a new theme template to CLI. (#1143)

1.修改主题切换逻辑，默认值改成从注册表中的themes获取。
只有亮色和暗色两种默认主题的情况下，保持原来的逻辑，点击直接切换主题
用户添加了自定义主题，出现选择列表，在列表中切换主题
2.添加新主题模板
在终端中，用命令创建新主题，选择theme类型，添加主题模板，输入主题名称：demo-theme
在demo-theme下可以看到新创建的主题，cd demo-theme && pnpm i
将主题接入设计器

gene9831 committed:
feat: add ESLint TypeScript rules and fix all ESLint errors (#1145)

项目ts改造基础，使用eslint9+typescript。在@vue/eslint-config-typescript默认配置上进行修改
